### PR TITLE
0.22.0 backports

### PR DESCRIPTION
**Pull request by TomAugspurger**

### CODE REVIEW

**Comment by TomAugspurger** [comment]
cc @jreback

**Comment by jreback** [comment]
lgtm. hopefully will backport cleanly :>

**Comment by TomAugspurger** [comment]
A couple builds on https://travis-ci.org/pandas-dev/pandas/builds/322879097 failed (which is just 0.21.1 + a new tag).

The error in https://travis-ci.org/pandas-dev/pandas/jobs/322879104 shouldn't be a problem since it's a conda-build thing which won't affect the wheels or conda-forge.

I'm haven't looked closely at the failures in https://travis-ci.org/pandas-dev/pandas/jobs/322879098#L1824

**Comment by jreback** [comment]
yeah those might actually be real.

**Comment by TomAugspurger** [comment]
Oh, whoops, I used the `td.skip_if_no` decorator that wasn't backported (`nanprod` was used in 1.10.0). I'll just push a commit here that does it manually.

I'll look into those other failures too.

**Comment by TomAugspurger** [comment]
I messed up the backport earlier since a `test_categorical.py` was split. Should be fixed now.

File: pandas/tests/test_categorical.py [hunk 1]
```diff
@@ -3163,18 +3163,6 @@ def test_info(self):
         buf = compat.StringIO()
         df2.info(buf=buf)
 
-    def test_groupby_sort(self):
```
Comment (TomAugspurger) [inline pandas/tests/test_categorical.py]: This and test_groupby were moved to `tests/categorical/test_groupby.py`. `test_groupby` was updated for the API change.

**Comment by jreback** [comment]
you might need to add: https://github.com/pandas-dev/pandas/commit/9c1071be085153d2b7d06417f5b2056ed86a8a70 to make the CI pass.

**Pull request merged**